conn.go

```diff
@@ -59,17 +65,60 @@ func (cc *Conn) Flush() error {
 
 	defer conn.Close()
 
-	if _, err := conn.SendMessages(batch(cc.messages)); err != nil {
+	smsg, err := conn.SendMessages(batch(cc.messages))
+
+	if err != nil {
 		return fmt.Errorf("SendMessages: %w", err)
 	}
 
-	if _, err := conn.Receive(); err != nil {
-		return fmt.Errorf("Receive: %w", err)
+	// Retrieving of seq number associated to entities
+	entitiesBySeq := make(map[uint32]Entity)
+	for i, e := range cc.entities {
+		entitiesBySeq[smsg[i].Header.Sequence] = e
+	}
+
+	// Trigger entities callback
+	for cc.checkReceive(conn) {
+		rmsg, err := conn.Receive()
+
+		if err != nil {
+			return fmt.Errorf("Receive: %w", err)
+		}
+
+		for _, msg := range rmsg {
+			if e, ok := entitiesBySeq[msg.Header.Sequence]; ok {
+				e.HandleResponse(msg)
+			}
+		}
 	}
 
 	return nil
 }
 
+func (cc *Conn) checkReceive(c *netlink.Conn) bool {
+	if cc.TestDial != nil {
+		return false
+	}
+
+	sc, err := c.SyscallConn()
+
+	var n int
+
+	sc.Control(func(fd uintptr) {
```

Comment:
I have not played with netlink's advanced features, so maybe I am overreacting but this comment from netlink library seem to me concerning:

> Once invoked, it is the caller's responsibility to ensure that operations performed using Conn and the syscall.RawConn do not conflict with each other.

How do you address/prevent it? What are the consequences of possible conflict?

Reply:
I think it's because operations with conn use mutex to prevent concurrency. There's a mutex in Flush so it's looks good to me.